File: backend/src/main/java/com/drivehub/vehiclerental/service/VehicleService.java
package com.drivehub.vehiclerental.service;

import com.drivehub.vehiclerental.dto.VehicleRequest;
import com.drivehub.vehiclerental.dto.VehicleResponse;

import java.util.List;

/**
 * Service interface for vehicle operations
 */
public interface VehicleService {
    
    /**
     * Create a new vehicle
     * 
     * @param request Vehicle request
     * @return Created vehicle response
     */
    VehicleResponse createVehicle(VehicleRequest request);
    
    /**
     * Get vehicle by ID
     * 
     * @param id Vehicle ID
     * @return Vehicle response
     */
    VehicleResponse getVehicleById(Long id);
    
    /**
     * Get all vehicles
     * 
     * @return List of vehicle responses
     */
    List<VehicleResponse> getAllVehicles();
    
    /**
     * Get available vehicles
     * 
     * @return List of available vehicle responses
     */
    List<VehicleResponse> getAvailableVehicles();
    
    /**
     * Update vehicle
     * 
     * @param id Vehicle ID
     * @param request Vehicle request
     * @return Updated vehicle response
     */
    VehicleResponse updateVehicle(Long id, VehicleRequest request);
    
    /**
     * Delete vehicle
     * 
     * @param id Vehicle ID
     */
    void deleteVehicle(Long id);
    
    /**
     * Update vehicle availability
     * 
     * @param id Vehicle ID
     * @param availability Availability status
     * @return Updated vehicle response
     */
    VehicleResponse updateAvailability(Long id, Boolean availability);
}

File: frontend/src/services/vehicleService.js
import axiosInstance from './axiosInstance';

// Vehicle Service
export const vehicleService = {
  // Get all vehicles
  getAllVehicles: async () => {
    const response = await axiosInstance.get('/vehicles');
    return response.data;
  },

  // Get available vehicles
  getAvailableVehicles: async () => {
    const response = await axiosInstance.get('/vehicles/available');
    return response.data;
  },

  // Get vehicle by ID
  getVehicleById: async (id) => {
    const response = await axiosInstance.get(`/vehicles/${id}`);
    return response.data;
  },

  // Create vehicle (Admin only)
  createVehicle: async (vehicleData) => {
    const response = await axiosInstance.post('/vehicles', vehicleData);
    return response.data;
  },

  // Update vehicle (Admin only)
  updateVehicle: async (id, vehicleData) => {
    const response = await axiosInstance.put(`/vehicles/${id}`, vehicleData);
    return response.data;
  },

  // Delete vehicle (Admin only)
  deleteVehicle: async (id) => {
    const response = await axiosInstance.delete(`/vehicles/${id}`);
    return response.data;
  },

  // Update vehicle availability (Admin only)
  updateAvailability: async (id, availability) => {
    const response = await axiosInstance.patch(
      `/vehicles/${id}/availability?availability=${availability}`
    );
    return response.data;
  },
};

export default vehicleService;
